ap.md

```diff
@@ -0,0 +1,256 @@
+    showIndex(urlParts.pathname, request, response);
+  }
+});
+
+// Stays the same
+server.listen(8080);
+
+// Stays the same
+console.log('Server running at http://192.168.1.101:8080/');
+{% endhighlight %}
+
+Inside our `createServer` callback function, we've replaced the old plain text response with a basic router that catches requests to '/leds/:index' and defaults all other requests to a separate function. Let's add those functions below the `console.log`:
+
+{% highlight javascript %}
+// Respond to the request with our index.html page
+function showIndex (url, request, response) {
+  response.writeHead(200, {"Content-Type": "text/html"});
+  // Use fs to read in index.html
+  fs.readFile(__dirname + '/index.html', function (err, content) {
+    if (err) {
+      throw err;
+    }
+
+    // Serve the content of index.html read in by fs
+    response.end(content);
+  });
+}
+
+// Toggle the led specified in the url and respond with its state
+function toggleLED (url, request, response) {
+  var indexRegex = /(\d)$/; // Create a regular expression to find the number at the end of the url
+  var result = indexRegex.exec(url); // Capture the number, returns an array
+  var index = +result[1]; // Grab the captured result from the array and make sure it's a Number
+
+  var led = tessel.led[index]; // Use the index to refence the correct LED
+
+  // Toggle the state of the led and call the callback after that's done
+  led.toggle(function (err) {
+    if (err) {
+      // Log the error and send back a 500 (internal server error) response to the client
+      console.log(err);
+      response.writeHead(500, {"Content-Type": "application/json"});
+      response.end(JSON.stringify({error: err}));
+    } else {
+      // The led was successfully toggled, so respond with the state of the toggled led using led.isOn
+      response.writeHead(200, {"Content-Type": "application/json"});
+      response.end(JSON.stringify({on: led.isOn}));
+    }
+  });
+}
+{% endhighlight %}
+
+One last step is to create a `.tesselinclude` file in the project directory and add any files not required by `ap.js` that we want to deploy with it. For this project, that means adding one line: 
+
+```
+index.html
+``` 
+
+Finally, let's fire up our server again by running:
+
+`t2 run ap.js`
+
+<div class="small-12 medium-6 columns">
+Just like before, once you see the "Server running at http://192.168.1.101:8080/" message in your terminal, you should be able to connect to that url in the web browser of the device connected to the TesselRouter network. After connecting, you should see the index.html view we built earlier and, after clicking/tapping the buttons, see either the blue or green LEDs on the Tessel should be toggled on or off. 
+</div>
+
+<div class="small-12 medium-6 columns">
+<video src="https://dl.dropboxusercontent.com/u/74986127/tessel-router-demo.mp4" controls loop class="small-12" >Video not available at the moment.</video>
```

Comment:
It ended up being too low resolution. Plus a video gives people the ability to fullscreen and control the video. Since this demo has a few moving parts to it, I think the video will be a better preview.